packages/react-native/scripts/codegen/generate-artifacts-executor.js

```diff
@@ -289,6 +294,33 @@ function findLibrariesFromReactNativeConfig(projectRoot) {
   });
 }
 
+// Function to look for libraries explicitly unlinked from the app
+// through the react-native.config.js file.
+// If this happens, it might be that the app does not need
+// to generate code for that library as it won't be used by that platform
+// @return { [libraryName: string]: [platform: string] }
+function findNotLinkedLibraries(projectRoot) {
+  const rnConfig = readRNConfigJSFile(projectRoot);
+
+  if (rnConfig.dependencies == null) {
+    return {};
+  }
+
+  let notLinkedLibraries = {};
+
+  Object.keys(rnConfig.dependencies).forEach(name => {
+    const dependency = rnConfig.dependencies[name];
+    let notLinkedPlatforms = [];
+    Object.keys(dependency.platforms).forEach(platform => {
+      if (dependency.platforms[platform] == null) {
+        notLinkedPlatforms.push(platform);
+      }
+    });
```

Comment:
Previously this was sufficient for module to be linked:

```js
  dependencies: {
    'name': {
      root: '/path/to/your/module',
    },
  },
```

With this change, projects with this config crash during `pod install` since the `platforms` field isn't present.

Reply:
Thanks for raising this other case. I believe that if there is just the path, it links for all the platforms, right?

Reply:
@cipolleschi currently it crashes when `dependency.platforms` isn't specified because the above check is accessing keys on `dependency.platforms` unconditionally